src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooBuildPlanService.java

```diff
@@ -202,11 +202,27 @@ private Stage createBuildStage(ProgrammingLanguage programmingLanguage, ProjectT
                 var tasks = readScriptTasksFromTemplate(programmingLanguage, "", sequentialBuildRuns, false, null);
                 tasks.add(0, checkoutTask);
                 defaultJob.tasks(tasks.toArray(new Task[0]));
+
                 // Final tasks:
                 final TestParserTask testParserTask = new TestParserTask(TestParserTaskProperties.TestType.JUNIT).resultDirectories("test-reports/*results.xml");
-                final ScriptTask cleanupTask = new ScriptTask().description("cleanup").inlineBody("sudo rm -rf tests/\nsudo rm -rf assignment/\nsudo rm -rf test-reports/");
-                defaultJob.finalTasks(testParserTask, cleanupTask);
+                defaultJob.finalTasks(testParserTask);
+
+                if (Boolean.TRUE.equals(staticCodeAnalysisEnabled)) {
```

Comment:
Is this verbose check really needed here? Wouldn’t a simple `if(staticCodeAnalysisEnabled )` be enough? (Depending on `staticCodeAnalysisEnabled` being a bool.)

Reply:
`staticCodeAnalysisEnabled` is actually a `Boolean` and not a primitive type. I am not 100% sure why `Boolean` was chosen here, but generally it should offer more useful features. `src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java`, where the variable is declared, only uses `Boolean` in general, so consistency might be another reason.